operator-framework-core/src/main/java/io/javaoperatorsdk/operator/processing/dependent/AbstractDependentResource.java

```diff
@@ -20,15 +18,15 @@
     implements DependentResource<R, P> {
   private static final Logger log = LoggerFactory.getLogger(AbstractDependentResource.class);
 
-  protected final boolean creatable = this instanceof Creator;
-  protected final boolean updatable = this instanceof Updater;
-  protected final boolean bulk = this instanceof BulkDependentResource;
+  private final boolean creatable = this instanceof Creator;
+  private final boolean updatable = this instanceof Updater;
+  private final boolean bulk = this instanceof BulkDependentResource;
 
   protected Creator<R, P> creator;
   protected Updater<R, P> updater;
-  protected BulkDependentResource<R, P> bulkDependentResource;
-
-  private final List<ResourceDiscriminator<R, P>> resourceDiscriminator = new ArrayList<>(1);
+  private final BulkDependentResource<R, P> bulkDependentResource;
+  private ResourceDiscriminator<R, P> resourceDiscriminator;
+  private int currentCount;
```

Comment:
Same problem here, what mentioned on other PR, that this cannot be a global variable